tests/Paysera/WalletApi/MapperTest.php

```diff
@@ -114,4 +126,152 @@ public function testDecodesPep()
         self::assertEquals('relationValue', $pepObj->getRelation());
         self::assertEquals('positionAValue', $pepObj->getPositions()[0]);
     }
+
+    /**
+     * @param $input
+     * @param $expected
+     * @return void
+     * @dataProvider decodeClientPermissionsToWalletDataProvider
+     */
+    public function testDecodeClientPermissionsToWallet($input, $expected)
+    {
+        self::assertEquals($expected, $this->mapper->decodeClientPermissionsToWallet($input));
+    }
+
+    /**
+     * @param $input
+     * @param $expected
+     * @return void
+     * @dataProvider encodeClientPermissionsToWalletDataProvider
+     * @throws Paysera_WalletApi_Exception_LogicException
+     */
+    public function testEncodeClientPermissionsToWallet($input, $expected)
+    {
+        if ($expected instanceof Exception) {
+            self::setExpectedException(get_class($expected), $expected->getMessage());
+        }
+        self::assertEquals($expected, $this->mapper->encodeClientPermissionsToWallet($input));
+    }
+
+    public function decodeClientPermissionsToWalletDataProvider()
+    {
+        return [
+            'case_1_empty_scopes' => [
```

Comment:
let's use not snake cased format, just write in words something like... 'No scopes'

Reply:
Done